src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -757,16 +754,12 @@ internal static void UnregisterMemoryLoadChangeNotification(Action notification)
 
 #endif
             }
-            else if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
-            {
-                ThrowHelper.ThrowInvalidTypeWithPointersNotSupported(typeof(T));
-            }
 
             GC_ALLOC_FLAGS flags = GC_ALLOC_FLAGS.GC_ALLOC_ZEROING_OPTIONAL;
             if (pinned)
                 flags |= GC_ALLOC_FLAGS.GC_ALLOC_PINNED_OBJECT_HEAP;
 
-            return Unsafe.As<T[]>(AllocateNewArray(RuntimeTypeHandle.ToIntPtr(typeof(T[]).TypeHandle), length, flags));
+            return Unsafe.As<T[]>(AllocateNewArray(typeof(T[]).TypeHandle.Value, length, flags));
```

Comment:
nit - I would just add a comment here to say that the runtime will always clear the `GC_ALLOC_ZEROING_OPTIONAL` flag if a type contains refs, just to make it clear that GC will always memory for types containing refs (which always needs to be enforced) so you don't need to filter it here.

Reply:
```suggestion
            // Runtime overrides GC_ALLOC_ZEROING_OPTIONAL if the type contains references, so we don't need to worry about that.
            return Unsafe.As<T[]>(AllocateNewArray(typeof(T[]).TypeHandle.Value, length, flags));
```
?